discovery.go

```diff
@@ -1,78 +1,42 @@
 package discovery
 
 import (
-	"errors"
 	"fmt"
 	"net/url"
 	"os"
 	"strings"
-
-	kv "gopkg.in/clever/kayvee-go.v2"
 )
 
-// m is a convenience type for using kv.
-type m map[string]interface{}
-
 const (
-	templateVar = "SERVICE_%s_%s_%%s"
+	templateVar = "SERVICE_%s_%s_URL"
 )
 
-func getVar(envVar string) (string, error) {
-	envVar = strings.ToUpper(envVar)
-	envVar = strings.Replace(envVar, "-", "_", -1)
-	val := os.Getenv(envVar)
+// URLString finds the specified URL for a service based off of the service's name and which
+// interface you are accessing. Values are found in environment variables fitting the scheme:
+// SERVICE_{SERVICE NAME}_{INTERFACE NAME}_URL.
+func URLString(service, name string) (string, error) {
+	// build string
+	envKey := fmt.Sprintf(templateVar, service, name)
+
+	// standardize to uppercase & underscores
+	envKey = strings.Replace(strings.ToUpper(envKey), "-", "_", -1)
+
+	val := os.Getenv(envKey)
 	if val == "" {
-		return "", errors.New(kv.FormatLog("discovery-go", kv.Error, "missing env var", m{
-			"var": envVar,
-		}))
+		return "", fmt.Errorf("Missing env var, %s", envKey)
```

Comment:
Do you think then that it would be the responsibility of the app using this library to catch the appropriate error and write a log with the correct `source, title`? These seem like important errors we would want to catch in metrics/alerting.

Another direction could be to wrap these lower level functions with a struct you initialize with an app name, list of URLs to get. That could contain the logic for getting all required env vars, writing the appropriate kayvee logs and returning a error to the app. (not convinced this is the right direction to go in, just curious about other people's opinion)